common/thorhelper/thorsoapcall.cpp

```diff
@@ -2523,19 +2549,25 @@ class CWSCAsyncFor : implements IWSCAsyncFor, public CInterface, public CAsyncFo
                             idx = 0;
                         if (idx==startidx)
                         {
+                            master->activitySpanScope->recordException(e, true, true);
                             StringBuffer s;
                             master->logctx.CTXLOG("Exception %s", e->errorMessage(s).str());
                             processException(url, inputRows, e);
                             return;
                         }
                     } while (blacklist->blacklisted(url.port, url.host));
+
+                    master->activitySpanScope->recordException(e, false, false); //Record the exception, but don't set failure
                 }
             }
             try
             {
                 checkTimeLimitExceeded(&remainingMS);
                 checkRoxieAbortMonitor(master->roxieAbortMonitor);
+                Owned<ISpan> socketOperationSpan = master->activitySpanScope->createClientSpan("Socket Write");
```

Comment:
Not convinced this span name is correct - but that can be revised later.  I tried looking at https://opentelemetry.io/docs/specs/semconv/http/http-spans/#method-placeholder, but it wasn't very helpful!